python/cuspatial/cuspatial/tests/test_geoseries.py

```diff
@@ -0,0 +1,313 @@
+            gs[gs.type == "Polygon"],
+            gs[gs.type == "MultiPolygon"],
+        ]
+    )
+    return result.reset_index(drop=True)
+
+
+def random_polygon(distance_from_origin):
+    outer = Point(distance_from_origin * 2, 0).buffer(1)
+    inners = []
+    for i in range(np.random.randint(1, 4)):
+        inner = Point(distance_from_origin + i * 0.1, 0).buffer(0.01)
+        inners.append(inner)
+    together = Polygon(outer, inners)
+    result = rotate(together, np.random.random() * 2 * np.pi, use_radians=True)
+    return result
+
+
+def random_multipolygon(size):
+    polygons = []
+    for i in range(size):
+        polygons.append(random_polygon(i))
+    result = MultiPolygon(polygons)
+    return result
+
+
+def generator(size, has_z=False):
+    obj_type = np.random.randint(1, 7)
+    if obj_type == 1:
+        return Point(np.random.random(2))
+    else:
+        if obj_type == 2:
+            points = np.random.random(size * 2).reshape(size, 2)
+            return MultiPoint(points)
+        elif obj_type == 3:
+            points = np.random.random(size * 2).reshape(size, 2)
+            return LineString(points)
+        elif obj_type == 4:
+            num_lines = np.random.randint(3, np.ceil(np.sqrt(size)) + 3)
+            points = np.random.random(num_lines * size * 2).reshape(
+                num_lines, size, 2
+            )
+            return MultiLineString(tuple(points))
+        elif obj_type == 5:
+            return random_polygon(size)
+        elif obj_type == 6:
+            return random_multipolygon(size)
+
+
+def to_shapely(obj):
+    if isinstance(obj, gpd.GeoSeries) and len(obj) == 1:
+        return obj[0]
+    if isinstance(obj, (cuPoint, cuLineString)):
+        return obj.to_shapely()
+    return obj
+
+
+def assert_eq_point(p1, p2):
+    p1 = to_shapely(p1)
+    p2 = to_shapely(p2)
+    assert type(p1) == type(p2)
+    assert p1.x == p2.x
+    assert p1.y == p2.y
+    assert p1.has_z == p2.has_z
+    if p1.has_z:
+        assert p1.z == p2.z
+    assert True
+
+
+def assert_eq_multipoint(p1, p2):
+    p1 = to_shapely(p1)
+    p2 = to_shapely(p2)
+    assert type(p1) == type(p2)
+    assert len(p1) == len(p2)
+    for i in range(len(p1)):
+        assert_eq_point(p1[i], p2[i])
+
+
+def assert_eq_linestring(p1, p2):
+    p1 = to_shapely(p1)
+    p2 = to_shapely(p2)
+    assert type(p1) == type(p2)
+    assert len(p1.coords) == len(p2.coords)
+    for i in range(len(p1.coords)):
+        assert_eq(p1.coords[i], p2.coords[i])
+
+
+def assert_eq_multilinestring(p1, p2):
+    p1 = to_shapely(p1)
+    p2 = to_shapely(p2)
+    for i in range(len(p1)):
+        assert_eq_linestring(p1[i], p2[i])
+
+
+def assert_eq_polygon(p1, p2):
+    p1 = to_shapely(p1)
+    p2 = to_shapely(p2)
+    if not p1.equals(p2):
+        raise ValueError
+
+
+def assert_eq_multipolygon(p1, p2):
+    p1 = to_shapely(p1)
+    p2 = to_shapely(p2)
+    if not p1.equals(p2):
+        raise ValueError
```

Comment:
Not sure that the individual `assert_eq_type` functions are being used anymore. They should be.

Reply:
They are.